### PR TITLE
First working stage

### PR DESCRIPTION
**Pull request by juliensnz**
[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

[//]: <> (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md)

**Description (for Contributor and Core Developer)**

[//]: <> (What does this Pull Request do? reference the related issue?)

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added Behats                      | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

**Comment by a2xchip** [comment]
Tada! What is the ticket?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/media-field.js [hunk 1]
```diff
@@ -103,8 +102,7 @@ define([
                     var message = xhr.responseJSON && xhr.responseJSON.message ?
                         xhr.responseJSON.message :
                         _.__('pim_enrich.entity.product.error.upload');
-                    navigation.addFlashMessage('error', message);
-                    navigation.afterRequest();
+                    messenger.addFlashMessage('error', message);
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/media-field.js]: should be `notificationFlashMessage()`

File: features/Context/AssertionContext.php [hunk 2]
```diff
@@ -570,12 +570,20 @@ public function theFollowingCodesShouldNotBeAvailable($entity, TableNode $table)
      */
     public function iShouldHaveNewNotification($count)
     {
-        $actualCount = (int) $this->getCurrentPage()->find('css', '.AknBell-countContainer')->getText();
+        $this->spin(function () use ($count) {
+            $actualCount = (int) $this->getCurrentPage()->find('css', '.AknBell-countContainer')->getText();
 
-        assertEquals(
-            $actualCount,
+            assertEquals(
```
Comment (pierallard) [inline features/Context/AssertionContext.php]: You can just do `return $count === $actualCount;`

File: features/Pim/Behat/Context/NavigationContext.php [hunk 3]
```diff
@@ -448,12 +445,13 @@ public function getCurrentPage()
...
-            $actualFullUrl = $this->getSession()->getCurrentUrl();
-            $actualUrl     = $this->sanitizeUrl($actualFullUrl);
-            $result        = parse_url($expected, PHP_URL_PATH) === $actualUrl;
-            assertTrue($result, sprintf('Expecting to be on page "%s", not "%s"', $expected, $actualUrl));
+          $actualFullUrl = $this->getSession()->getCurrentUrl();
+          $actualUrl     = $this->sanitizeUrl($actualFullUrl);
+          $result        = parse_url($expected, PHP_URL_PATH) === $actualUrl;
 
-            return true;
+          assertTrue($result, sprintf('Expecting to be on page "%s", not "%s"', $expected, $actualUrl));
```
Comment (pierallard) [inline features/Pim/Behat/Context/NavigationContext.php]: Same, you can do return $expected === $actualUrl (and put the sprintf into spin message

File: src/Oro/Bundle/UserBundle/Resources/translations/jsmessages.en.yml [hunk 4]
```diff
@@ -1 +1,25 @@
 "Update status": "Update status"
+page_title:
```
Comment (pierallard) [inline src/Oro/Bundle/UserBundle/Resources/translations/jsmessages.en.yml]: Please remove all the non js translation related to these ones

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/router.js [hunk 5]
```diff
@@ -65,6 +65,11 @@ define(
                 if (path.indexOf('/') !== 0) {
                     path = '/' + path;
                 }
+
+                if (path.indexOf('|') !== -1) {
```
Comment (a2xchip) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/router.js]: `-1 !== path.indexOf('|')`

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/router.js [hunk 6]
```diff
@@ -86,13 +91,19 @@ define(
 
                     if (controller.aclResourceId && !securityContext.isGranted(controller.aclResourceId)) {
                         this.displayErrorPage(__('pim_enrich.error.http.403'), '403');
+
+                        return;
                     }
 
                     this.currentController = new controller.class({ el: $view});
```
Comment (a2xchip) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/router.js]: I do not see `controller` class implementation in current subset, but feel like it is not about controller at all but about action, may be naming is wrong?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/controller/association-type.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    function (_, BaseController, FormBuilder, FetcherRegistry, UserContext, Dialog, PageTitle, Error, i18n) {
+        return BaseController.extend({
+            /**
+             * {@inheritdoc}
+             */
+            renderRoute: function (route) {
+                return FetcherRegistry.getFetcher('association-type').fetch(route.params.code, {cached: false})
+                    .then(function (associationType) {
+                        if (!this.active) {
+                            return;
```
Comment (tamarasaurus) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/controller/association-type.js]: In which situation would the controller be inactive but still executing `renderRoute`? Does the fetch request need to run every time? (Same for the other controllers)

File: src/Pim/Bundle/UserBundle/Resources/public/js/init-user.js [hunk 8]
```diff
@@ -1,32 +1,40 @@
...
-        }
+define(['jquery', 'underscore', 'oro/translator', 'oro/app', 'oro/mediator', 'oro/messenger'],
+    function () {
+        // $, _, __, app, mediator, messenger
+
+        /* ============================================================
+         * from user.js
+         * ============================================================ */
+        return function () {
+            // function initFlashMessages() {
```
Comment (tamarasaurus) [inline src/Pim/Bundle/UserBundle/Resources/public/js/init-user.js]: Is this going to be completed or can we 🔥  ?

File: src/Pim/Bundle/UIBundle/Resources/public/js/init-layout.js [hunk 9]
```diff
@@ -124,22 +114,22 @@ require(['jquery', 'underscore', 'oro/translator', 'oro/app', 'oro/mediator', 'o
...
-     * ============================================================ */
-    $(document).on('click', '.action-button', function () {
-        var actionInput = $('input[name = "input_action"]');
-        actionInput.val($(this).attr('data-action'));
-        $('#' + actionInput.attr('data-form-id')).submit();
-    });
+        /* ============================================================
+         * from form_buttons.js
+         * ============================================================ */
+        $(document).on('click', '.action-button', function () {
```
Comment (tamarasaurus) [inline src/Pim/Bundle/UIBundle/Resources/public/js/init-layout.js]: Do we still need these global events ? I would hope that all the backbone views handle this logic

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/controller/user.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+        'pim/controller/form'
+    ], function (
+        FormController
+    ) {
+        return FormController.extend({
+            /**
+             * {@inheritdoc}
+             */
+            afterSubmit: function () {
+                location.reload(); //TODO nav: reload the page to update the menu
```
Comment (tamarasaurus) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/controller/user.js]: ❓

File: src/Pim/Bundle/UIBundle/Resources/public/js/init-layout.js (hunk 9 above)
Comment (juliensnz) [inline src/Pim/Bundle/UIBundle/Resources/public/js/init-layout.js]: we should burn them

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/controller/user.js (hunk 10 above)
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/controller/user.js]: yeah, when the screen will be reworked

**Pull request merged**